### PR TITLE
[Integration][Snyk] Catch error code SNYK-9999 from snyk api

### PR DESCRIPTION
**Pull request by omby8888**
# Description

What - catch synk internal error while fetching project issues

## Type of change

Please leave one option from the following and delete the rest:

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/snyk/snyk/client.py [hunk 1]
```diff
@@ -105,15 +106,24 @@ async def get_issues(self, org_id: str, project_id: str) -> list[dict[str, Any]]
...
+            event.attributes[cache_key] = issues
+            return issues
+        except httpx.HTTPStatusError as e:
+            if json.loads(e.response.text).code == "SNYK-9999":
+                logger.error(
+                    f"Encountered Synk internal error while getting issues "
+                    f"for project {project_id} in org {org_id}."
+                )
+                return []
+            raise
```
Comment (Tankilevitch) [inline integrations/snyk/snyk/client.py:126]: this error can occur on every request, maybe we should do it inside the `self._send_api_request`

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Missing changelog and version bump

File: integrations/snyk/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
+# Port_Ocean 0.1.33 (2024-03-31)
+
+### Bug Fixes
+
+- catch error code SNYK-9999 from snyk api
```
Comment (Tankilevitch) [inline integrations/snyk/CHANGELOG.md:14]:
```suggestion
- Fixed an issue when the snyk API returns internal error code SNYK-9999 and caused the resync to fail (PORT-7454)
```

File: integrations/snyk/snyk/client.py [hunk 3]
```diff
@@ -65,6 +67,16 @@ async def _send_api_request(
...
         except httpx.HTTPStatusError as e:
+            if (
+                raise_for_snyk_error
+                and json.loads(e.response.text).get("code") == "SNYK-9999"
+            ):
+                logger.error(
+                    f"Encountered Synk internal error while sending request: method: {method}, "
+                    f"url: {url}, query_params: {query_params}, version: {version}, json: {json_data}"
+                )
+                return {}
```
Comment (yairsimantov20) [inline integrations/snyk/snyk/client.py:78]: Why not failing for it?

File: integrations/snyk/snyk/client.py [hunk 4]
```diff
@@ -111,7 +123,11 @@ async def get_issues(self, org_id: str, project_id: str) -> list[dict[str, Any]]
                 method="POST",
                 version=self.snyk_api_version,
             )
-        )["issues"]
+        ).get("issues")
+
+        if not issues:
+            return []
```
Comment (yairsimantov20) [inline integrations/snyk/snyk/client.py:129]:
```suggestion
        ).get("issues", [])
```

File: integrations/snyk/snyk/client.py [hunk 5]
```diff
@@ -322,13 +345,16 @@ async def create_webhooks_if_not_exists(self) -> None:
...
+                method="POST",
+                json_data=body,
+                raise_for_snyk_error=True,
             )
 
     async def get_all_organizations(self) -> list[dict[str, Any]]:
         url = f"{self.api_url}/orgs"
         response = await self._send_api_request(url=url)
-        organizations = response["orgs"]
+        organizations = response.get("orgs") or []
```
Comment (yairsimantov20) [inline integrations/snyk/snyk/client.py:357]:
```suggestion
        organizations = response.get("orgs", [])
```

**Review by yairsimantov20 — CHANGES_REQUESTED** [review]

File: integrations/snyk/snyk/client.py (hunk 4 above)
Comment (omby8888) [reply integrations/snyk/snyk/client.py:129]: I don't want [] to be cached so i return it immediately

**Review by Tankilevitch — APPROVED** [review]
Looks good, I would suggest adding a hint in our port-docs snyk docs that we are currently skipping cases where experiencing SNYK-9999 errors

**Pull request merged**